### ISSUE TITLE
Do not set the current working directory to the user home

### ISSUE THREAD

**Issue by ChristianCiach** [signageos/vscode-sops#43]
Currently, when calling the `sops` binary, this extension changes the working directory to `$HOME`. I would like to use the current *workspace* directory instead.

My use-case: Instead of letting your extension call the `sops`-binary directory, I am configuring `sops.binPath` to use a custom `sops-wrapper.sh`. My wrapper script decrypts an ansible-vault encrypted `keys.txt` that is found inside the workspace directory and finally sets the environment-variable `SOPS_AGE_KEY_FILE` before calling the actual `sops` binary. Something like that:

```bash
#!/bin/bash
ansible-vault decrypt --output /tmp/decrypted-sops-key.txt sops-key.txt
export SOPS_AGE_KEY_FILE=/tmp/decrypted-sops-key.txt 
exec sops "$@"
```

Currently I have no way to correctly reference the `sops-key.txt` of the workspace directory. There is also no existing env-variable that I could use to determine the current project directory.

On a similar note, it would be nice if `sops.binPath` could be set to a file found inside the workspace directory. VSCode provides the variable `${workspaceFolder}`, but it cannot be used when configuring `sops.binPath`.

I am sure there are other ways to solve my use-case, but I cannot think of any other way that doesn't involve changes to this vscode extension.

**Comment by ChristianCiach** [comment]
> On a similar note, it would be nice if sops.binPath could be set to a file found inside the workspace directory. VSCode provides the variable ${workspaceFolder}, but it cannot be used when configuring sops.binPath.

Actually, this may implicitly work if `spawnOptions.cwd` would point to the workspace directory. Then I could probably configure `sops.binPath` as `./sops-wrapper.sh` inside my workspace-settings.

**Comment by kitos9112** [comment]
I'm facing a similar issue in which I cannot decrypt age-encrypted files without amending the default macOS application support path ($HOME/Library/Application Support/sops/age/keys.txt)- not entirely sure why it does not respect the `SOPS_AGE_KEY_FILE` env. variable.

Also, my understanding is that this extension would have stuck with the `age` XDG standard so that if there's an env variable ($XDG_CONFIG_HOME) present, it would default to there as per the docs --> https://github.com/mozilla/sops#encrypting-using-age

**Comment by misak113** [comment]
Is it possible, that the problem of not using XDG_CONFIG_HOME is related to the specific logic under the hood of the extension which is using `/tmp/` folder for decryption/encryption? It's because of cross-platform compatibility for "Remote development".
So the real HOME directory maybe cannot be determined in this case.

**Comment by ChristianCiach** [comment]
I think the main issue is this piece of code:

```ts
let spawnOptions: child_process.SpawnSyncOptions = {
	cwd: process.env.HOME,
};
```
Why do you need to change the `cwd` instead of just leaving it at default?

**Comment by misak113** [comment]
The reason is probably that the files are encrypted and decrypted in a temporary folder (probably `/tmp`). And this should move you to the home directory and use all user settings in the home, like ~/.aws, etc.
I can try it out and we will see.

**Comment by cowens-genvidtech** [comment]
I agree this is an important feature. On a project I'm working on, we have a setup step for new team members to manually download SOPS and configure their environment so this extension can find it. We could eliminate this setup step by just including sops.exe in the repository and telling the extension to use SOPS from a path like ./sops.exe or ${workspaceRoot}/sops.exe in the shared VSCode configuration file we include in the repository's .vscode folder.
